requests/utils.py

```diff
@@ -90,6 +86,19 @@ def super_len(o):
             # let requests chunk it instead.
             current_position = total_length
 
+    if hasattr(o, 'seek') and total_length is None:
+        # StringIO has a notimplemented fileno
+        # BytesIO has seek and not fileno
+        total_length = o.seek(0, 2)
+        # StringIO objects in py2 returns None
+        if total_length is None and hasattr(o, 'tell'):
+            total_length = o.tell()
+        elif total_length is None:
+            total_length = 0
+
```

Comment:
Ah, ok. In that case, let's not rely on this at all because it confuses the control flow a bit. Let's just use `tell()`.